code/modules/mob/living/carbon/xenomorph/xenoprocs.dm

```diff
@@ -1,4 +1,39 @@
 /mob/living/carbon/xenomorph/Bump(atom/A)
+	if(ismecha(A))
+		var/obj/vehicle/sealed/mecha/mecha = A
+		var/mob_swap_mode = NO_SWAP
+		if(a_intent == INTENT_HELP)
+			mob_swap_mode = SWAPPING
+		// If we're moving diagonally, but the mob isn't on the diagonal destination turf and the destination turf is enterable we have no reason to shuffle/push them
+		if(moving_diagonally && (get_dir(src, mecha) in GLOB.cardinals) && get_step(src, dir).Enter(src, loc))
+			mob_swap_mode = PHASING
+		if(mob_swap_mode)
+			//switch our position with mech
+			if(loc && !loc.Adjacent(mecha.loc))
+				return
+			now_pushing = TRUE
+			var/oldloc = loc
+			var/oldmechaloc = mecha.loc
+
+			var/mecha_passmob = (mecha.allow_pass_flags & PASS_MOB) // we give PASS_MOB to both mobs to avoid bumping other mobs during swap.
+			var/src_passmob = (pass_flags & PASS_MOB)
+			mecha.allow_pass_flags |= PASS_MOB
+			pass_flags |= PASS_MOB
```

Comment:
we don't actually need to give the mob pass_mob afaik, should work with just the mech having allow_pass_flags |= PASS_MOB